### PR TITLE
Add segment support to config.yml

### PR DESCRIPTION
**Pull request by mknowlton89**
### Features and Changes

Currently, dimensions are included in config.yml, but segments are not.

With this PR, organizations will be able to add segments to their config.yml file and use them within their GB account.

Additionally, the PR adds support for a new environment variable `STORE_SEGMENTS_IN_MONGO` to handle some edge cases. Essentially, organizations that use the config.yml file today, likely have created segments that are currently stored in Mongo. When this change is merged in, if they attempt to use segments, it will look like they don't have any due to the fact that in `SegmentModel` if an org is using a config.yml file, we don't go to Mongo to fetch their segments, but instead go to the config file to fetch their segments.

With this new environment variable, an org can essentially say - yes, we use the config.yml file, but we want to continue using the segments stored in Mongo.

One use case that we aren't solving for in this PR is in the event an organization has segments in both the config.yml file AND Mongo, and has the `STORE_SEGMENTS_IN_MONGO` environment variable set to true. In that case (which currently doesn't affect any organizations) we will ALWAYS prioritize the segments in Mongo. We will not concatenate the segments in the config.yml file with the segments in Mongo.

If any orgs get into this state, we will instruct them to use one or the other, or revisit the logic.

Additionally, 

- Closes **(https://github.com/growthbook/growthbook/issues/780)**

### Dependencies

~~This PR is blocked by the SegmentModel refactor - https://github.com/growthbook/growthbook/pull/1016 and can only be merged in after that PR is merged.~~

### Testing && Screenshots

- [x] Normal flow 1, no config file, no mongo records - should see standard empty state & button
Screen Shot 2023-02-23 at 3 21 43 PM

- [x] Normal flow 2, no config file, mongo records - should see records
Screen Shot 2023-02-23 at 3 21 28 PM

- [x] Happy path 1 - new user, uploads config file without segments, ensure message is displayed that says “to add these, you can add them to the config file, or add the STORE_SEGMENTS_IN_MONGO env variable. No create button until that env variable is added.
Screen Shot 2023-02-23 at 3 17 37 PM

- [x] Happy path 2 - new user, uploads config file with segments, doesn't have STORE_SEGMENTS_IN_MONGO env variable, they should see them, but not see a create button.
Screen Shot 2023-02-23 at 3 16 18 PM

- [x] Rocky path 1 - existing user, that has a config file without segments, but has segments in Mongo, but doesn’t have the “STORE_SEGMENTS_IN_MONGO file - they should see “no segments because your config file doesn’t have any. Either add them to the config, or add the env variable. Essentially, this should be equal to happy path 1.
Screen Shot 2023-02-23 at 3 14 57 PM

- [x] Rocky path 2 - existing user, has a config file with segments, and also has segments in Mongo, has the env variable STORE_SEGMENTS_IN_MONGO so they should only see segments within Mongo.
Screen Shot 2023-02-23 at 3 11 47 PM

- [x] Rocky path 3 (edge case) - existing user, has a config file without segments, has the STORE_SEGMENTS_IN_MONGO env variable, but doesn't have any segments in Mongo. Should see a message to either add segments via the button, or add the segments to config file and remove STORE_SEGMENTS_IN_MONGO env variable
Screen Shot 2023-02-23 at 3 20 35 PM

### CODE REVIEW

**Review by bttf — DISMISSED** [review]
Looks good to me, perhaps another approval could verify

File: packages/back-end/src/models/SegmentModel.ts [hunk 1]
```diff
@@ -56,5 +83,11 @@ export async function updateSegment(
   organization: string,
   updates: Partial<SegmentInterface>
 ) {
+  // If using config.yml & the org doesn't have the env variable STORE_SEGMENTS_IN_MONGO,
+  // immediately return the list from there
+  if (usingFileConfigForSegments()) {
```
Comment (bryce-fitzsimons) [inline packages/back-end/src/models/SegmentModel.ts:88]: minor: should we also have something in the UI that shows why we can't add/remove segments when using a config file?

**Review by bryce-fitzsimons — DISMISSED** [review]
looks good, minor suggestions (feel free to ignore)

File: packages/back-end/src/models/SegmentModel.ts (hunk 1 above)
Comment (mknowlton89) [reply packages/back-end/src/models/SegmentModel.ts:89]: @bryce-fitzsimons Good rec - I just created an issue for that [here](https://github.com/growthbook/growthbook/issues/1051) as a follow-up.

**Review by bttf — APPROVED** [review]

**Pull request merged**